### PR TITLE
Move key funcs to app library

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/11656

Moving the key funcs so they can be used in a new validation package.

Also moving annotations since they are used in the key package.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+module github.com/giantswarm/app/v3
 
-go 1.14
+go 1.15
 
 require (
-	github.com/giantswarm/apiextensions/v2 v2.0.0
-	github.com/giantswarm/microerror v0.2.0
-	k8s.io/apimachinery v0.18.5
+	github.com/giantswarm/apiextensions/v3 v3.4.2-0.20201103105314-dd1460e43e4f
```
Comment (rossf7) [inline go.mod:6]: TODO: Needs https://github.com/giantswarm/apiextensions/pull/533

File: pkg/app/app.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 	"io"
 
-	applicationv1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/application/v1alpha1"
+	applicationv1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
```
Comment (rossf7) [inline pkg/app/app.go:8]: Bumping apiextensions to v3.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,12 +1,13 @@
-version: 2
-jobs:
-  build:
-    machine: true
-    steps:
-    - checkout
+version: 2.1
+orbs:
+  architect: giantswarm/architect@0.16.0
```
Comment (rossf7) [inline .circleci/config.yml:3]: Circle setup was very outdated.

Using latest orb means we need replaces for the security issues found by Nancey.

**Review by oponder — APPROVED** [review]
LGTM

**Review by tomahawk28 — APPROVED** [review]

**Pull request merged**